src/main/groovy/org/transmartproject/batch/highdim/mrna/data/MrnaDataJobConfiguration.groovy

```diff
@@ -59,10 +59,9 @@ class MrnaDataJobConfiguration extends AbstractStandardHighDimJobConfiguration {
                 partitionByColumn: 'trial_name',
                 partitionByColumnValue: studyId,
                 sequence: Sequences.MRNA_PARTITION_ID,
-                indexes: [
-                        ['assay_id'],
-                        ['probeset_id'],
-                        ['patient_id'],
+                primaryKey: ['assay_id', 'probeset_id'],
+                foreignKeys: [
+                        assay_id: [table: Tables.SUBJ_SAMPLE_MAP, column: 'assay_id', onDelete: 'CASCADE']
```

Comment:
`probeset_id` is part of primary key. Hence it will be part of composite index. I'll have a look with psql `EXPLAIN` how it's going to work.